sstmac/software/process/operating_system.cc

```diff
@@ -660,9 +661,23 @@ OperatingSystem::outcastAppStart(int my_rank, int aid, const std::string& app_ns
   for (int r=0; r < num_to_send; ++r){
     int dst_rank = ranks[r];
     int dst_nid = mapping->rankToNode(dst_rank);
+
+#if SSTMAC_INTEGRATED_SST_CORE
+    // using sst-core so we can't serialize params, will send a json string instead
+    std::set<std::string> keys = app_params.getKeys();
+    nlohmann::json j;
+    for (auto &k : keys) {
+        j[k] = app_params.find<std::string>(k,"not found");
+      }
     sw::StartAppRequest* lev = new StartAppRequest(node_->allocateUniqueId(),
                                      aid, app_ns, mapping, dst_rank, dst_nid,
-                                     addr(), app_params);
+                                     addr(), j.dump().c_str());
```

Comment:
This seems dangerous, dump creates a temp and c_str() returns a ptr to that temp.  I thought that you asked `StartAppRequest` to take a `std::string` so why pass in `c_str`? 

I don't want to actually try to think about how this works, in this case a `const char*`to a temporary is passed to a `std::string const&` and apparently works.  Life time extension is weird, but we shouldn't count on this working. Just passing in `j.dump()` should be okay.

Reply:
We might need to change the AppRequest to take a string since the lifetime issues seem complicated.  We probably guarantee that `app_params` lives long enough to work, but I'm not confident that the `j.dump()` string does.  I suppose I should actually figure out if lifetime extension works as expected in this case.

Reply:
Likely just a remnant from earlier version of fix that I didn't notice since it worked.

Reply:
Remvoed c_str() in cs 5841f71.